### DISCUSSION TITLE
Compiled version for Windows

### DISCUSSION THREAD

**Discussion by edumarzolla** [elalish/manifold#359] [Ideas]
Hi,
Looking for potential users that are not familiar with C++ or CUDA compillers, but need to add/subtract/intersec millions of primitives, I would like to suggest a compiled exe version for Windows64, where the user could for example execute the command:
manifoldcad.exe -f file.txt mesh.glb      (where the file.txt is a java file like the file below)

file.txt
const box = cube([100, 100, 100], true);
const ball = sphere(60, 100);
const result = box.subtract(ball);

Thank you

**Comment by pca006132** [comment]
we can provide prebuilt binary, but interpreting a java like script is likely out-of-scope for us.

**Reply by edumarzolla** [reply]
thanks for answering quickly...
how to use this prebuilt binary?
The objective is to create multiple cylinders in different posicions androtations... and subtract them

**Reply by elalish** [reply]
In addition to python, have you looked at https://manifoldcad.org/? If that looks promising, it would be easy to set it up for batched operations in Node.js.

**Reply by edumarzolla** [reply]
Hi,
Could you please provide that prebuild binaries?(preferable x64 TBB and x64 CUDA)

**Reply by elalish** [reply]
I think providing prebuilt binaries for Windows will be a large maintenance burden for us. If you want to contribute automation for that, PRs are welcome. Otherwise I'd recommend you build them locally, or use our Python library which is already pre-built for Windows and such. Also, we no longer support CUDA, so it's either single-threaded, or TBB. CUDA wasn't giving us much benefit over TBB, and was too hard to build.

**Reply by pca006132** [reply]
Our python binding is available on pypi: https://pypi.org/project/manifold3d/

**Reply by edumarzolla** [reply]
Hello... I really need help compiling Manifold into a DLL that can be used by other WinForms programs, for example, C# 64-bit.
Could you please help me with a tutorial?

**Reply by pca006132** [reply]
see #743

**Comment by pca006132** [comment]
We do have python, wasm and C binding, if you want to write scripts maybe you can try the python binding?

**Reply by edumarzolla** [reply]
I am trying to use the Python version 3.10, my sysop is Windows10 64.
I tryed: 
pip install pymanifold     -->success
but, when my code wun "from pymanifold import manifold", an error apear: No module named 'dreal'

when I try to install dreal via pip install dreal, another error apear:  ModuleNotFoundError: No module named 'wheel.pep425tags'
when I try to install wheel.pep425tags, pip says there is no distribution found for wheel.pep425tags

Any suggestion to get the pymanifold working in a Windows10 PC?

thanks

**Reply by pca006132** [reply]
Ah, `pymanifold` is a different module, we should probably document this better...
We have not yet published to PyPI, as the python binding is not feature complete (missing the latest features) and we are not familiar with PyPI. Currently, you can only compile from source which should not be hard if you have CMake and VisualStudio installed. We have CI setup for windows: https://github.com/elalish/manifold/blob/master/.github/workflows/manifold.yml#L145

Basically, you should get CMake and Visual Studio with C++ compiler, and then run the following commands under your directory:
```
cmake . -DCMAKE_BUILD_TYPE=Release -B build -DMANIFOLD_DEBUG=ON -DMANIFOLD_PYBIND=ON -DMANIFOLD_PAR=NONE -A x64
cd build
cmake --build . --target ALL_BUILD --config Release
```
And then there should be a `pymanifold.dll` or something in `build/bindings/python/`, which you can put in your python path and use. I am not sure how to install TBB on windows, so this compiles with no multithreading.

Alternatively, you can try WSL2 which iirc have pretty good performance.

**Comment by edumarzolla** [comment]
I installed the Ubuntu 22.10 64bits in a Virtual Machine and I got success
compiling the test.
My key objective is to develop a mesh resulted of 50.000 cylinders
subtracted from one bigger cylinder. All the cylinders must be positioned
and rotated according to a table. All these cylinders are parametrized
(size, XYZ positioning, rotation of the smaller cylinders)
If it can be done by a DLL that could be called by a Windows program (for
example VisualBasic), then we would have the problem solved - or python...


Here made via ManifoldCad frontend - that cannot subtract many cylinders
[image: image.png]

Here and example with AutoCAD
[image: image.png]

Em ter., 14 de mar. de 2023 às 02:36, Emmett Lalish <
***@***.***> escreveu:
